src/Corax/IndexWriter.cs

```diff
@@ -1464,15 +1497,15 @@ private unsafe void InsertSpatialField(Tree entriesToSpatialTree, IndexedField i
                 list.Sort();
 
                 ref var entryTerms = ref GetEntryTerms(entry);
-                for (int i = 0; i < list.Count; i++)
+                foreach (var item in list)
```

Comment:
Maybe  `CollectionsMarshal.AsSpan()` for that list?
Bench: https://www.meziantou.net/fastest-way-to-enumerate-a-list-t.htm

Reply:
Done